CONTRIBUTING.md

```diff
@@ -0,0 +1,158 @@
+- NVIDIA cuOpt service API access and credentials
+
+### Create the build environment
+
+1. Clone the repository:
+```bash
+git clone https://github.com/NVIDIA/cuopt-examples.git
+cd cuopt-examples
+```
+
+2. Create and activate a conda environment:
+```bash
+conda create -n cuopt-dev python=3.8
+conda activate cuopt-dev
+```
+
+3. Install dependencies:
+```bash
+pip install -r requirements.txt
+```
+
+4. Set up your NVIDIA cuOpt service API credentials:
+```bash
+export CUOPT_API_KEY=your_api_key_here
+```
+
+## Code Formatting
+
+### Using pre-commit hooks
+
+The project uses pre-commit to execute code linters and formatters. These tools ensure a consistent code format throughout the project. Using pre-commit ensures that linter versions and options are aligned for all developers.
+
+To use `pre-commit`, install via `conda` or `pip`:
+
+```bash
+conda install -c conda-forge pre-commit
+# or
+pip install pre-commit
+```
+
+Then run pre-commit hooks before committing code:
+
+```bash
+pre-commit run
+```
+
+By default, pre-commit runs on staged files. To run pre-commit checks on all files, execute:
+
+```bash
+pre-commit run --all-files
+```
+
+Optionally, you may set up the pre-commit hooks to run automatically when you make a git commit:
+
+```bash
+pre-commit install
+```
+
+You can skip these checks with `git commit --no-verify` or with the short version `git commit -n`.
+
+### Summary of pre-commit hooks
+
+The following section describes some of the core pre-commit hooks used by the repository. See `.pre-commit-config.yaml` for a full list.
+
+- Python code is formatted with Black
+- Imports are sorted with isort
+- Code is checked with flake8
+- Type hints are checked with mypy
+- Spelling is checked with codespell
+
+## Developer Guidelines
+
+### Python Code Style
+
+- Follow PEP 8 guidelines
+- Use type hints for function arguments and return values
+- Write docstrings for all public functions and classes
+- Keep functions focused and single-purpose
+- Use meaningful variable and function names
+- Add comments for complex logic
+
+### Example and Workflow Guidelines
+
+- Each example should be self-contained and runnable
+- Include clear documentation and comments
+- Add performance benchmarks where appropriate
+- Include error handling and edge cases
+- Provide example input data or instructions for generating test data
+- Add visualization of results where applicable
+- Follow the NVIDIA cuOpt service API best practices and guidelines
+- Include example configuration files for different use cases
+- Document any specific requirements or limitations
+
+### Notebook Guidelines
+
+- Use Jupyter notebooks for interactive examples
+- Include markdown cells with clear explanations
+- Add code comments for complex operations
+- Include visualization cells for results
+- Provide example outputs and expected results
+- Add troubleshooting sections for common issues
+- Include links to relevant documentation
+
+### Testing Guidelines
```

Comment:
Lets update this, since this doesn't make sense for notebooks. May be test locally and clear any outputs from notebook before it gets pushed.